app/controllers/developers_controller.rb

```diff
@@ -4,7 +4,35 @@ def index
     @developers = Developer.all
   end
 
+  def create
+    @developer = Developer.new(developer_constructor_params)
+    if @developer.save
+      flash[:notice] = "Logged in as #{@developer.name}"
+      session[:developer_id] = @developer.id
+      redirect_to developer_path
+    else
+      flash[:error] = @developer.errors.full_messages.join(", ")
+    end
+  end
+
   def show
-    @developer = Developer.find(params[:id])
+    if params[:id]
+      @developer = Developer.find(params[:id])
+    elsif current_developer
+      @developer = current_developer
+      render :current_developer
+    else
+      render file: 'public/404.html'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.